R/ingest-scp.R

```diff
@@ -238,30 +198,24 @@ get_incidence_for_all_inputs <- function(state, input_combinations) {
 #'
 
 write_to_db <- function(df, df_name) {
-  db_host <- Sys.getenv("DB_HOST")
-  db_name <- Sys.getenv("DB_NAME")
-  db_user <- Sys.getenv("DB_USER")
-  db_password <- Sys.getenv("DB_PASSWORD")
-  db_port <- Sys.getenv("DB_PORT")
-
   db_connection <- DBI::dbConnect(
     RPostgres::Postgres(),
-    host = db_host,
-    dbname = db_name,
-    user = db_user,
-    password = db_password,
-    port = db_port
+    host = Sys.getenv("DB_HOST"),
+    dbname = Sys.getenv("DB_NAME"),
+    user = Sys.getenv("DB_USER"),
+    password = Sys.getenv("DB_PASSWORD"),
+    port = Sys.getenv("DB_PORT")
   )
 
+  on.exit(DBI::dbDisconnect(db_connection))
+
```

Comment:
Best practice in R is that as soon as you create a db connection within a function, you ensure that you close the connection using `on.exit()`, which runs an expression when the function finished running "no matter what," even if there's an error. For example:

```r
add_two_numbers <- function(x, y){
  on.exit(cat("Exit!"))
  as.character(x) + as.character(y)
}

add_two_numbers(4, 5)

#> Error in as.character(x) + as.character(y) : 
#>   non-numeric argument to binary operator
#> Exit!
```